pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -59,10 +63,59 @@ func (a *v1beta1PolicyApplier) JwtFilter(isXDSMarshalingToAnyEnabled bool) *http
 	return out
 }
 
+// All explaining code link can be removed before merging.
+func convertToIstioAuthnFilterConfig(jwtRules []*v1beta1.JWT) *authn_filter.FilterConfig {
+	p := authn_alpha.Policy{
+		// Targets are not referenced in proxy repo.
+		// https://github.com/istio/proxy/search?q=targets&unscoped_q=targets
+		OriginIsOptional: true,
+		// Peers not really needed, always invoke the
+		// https://github.com/istio/proxy/blob/24e971ff31c5cce22e9ab49f8478629f50664846/src/envoy/http/authn/http_filter.cc#L84
+		PeerIsOptional: true,
+		// Always bind request.auth.principal from JWT origin. In v2 policy, authorization config specifies what principal to
+		// choose from instead, rather than in authn config.
+		PrincipalBinding: authn_alpha.PrincipalBinding_USE_ORIGIN,
+	}
+	// TODO: does it matter if this is empty but not nil?
+	jwtLocation := map[string]string{}
```

Comment:
I think this can be deprecated now. It was used to find the payload, but since we moved to dynamic metadata, the field was obsolete and unused.

Reply:
done.